wp-includes/functions.php

```diff
@@ -5736,19 +5736,36 @@ function wp_validate_boolean( $var ) {
 /**
  * Delete a file
  *
- * @since 4.2.0
+ * @since 4.9.2
  *
- * @param string $file The path to the file to delete.
+ * @param string $file                The path to the file to delete.
+ * @param array  $directories_unblock Are directory to unblocked. They are ROOT, WP_CONTENT_DIR, WP_ADMIN_DIR
+ *                                   and WP_INCLUDE_DIR.
+ * @return bool Whether the param is invalidated.
  */
-function wp_delete_file( $file ) {
+function wp_delete_file( $file, $directories_unblock = array() ) {
+
+	// Default Filters the path of the file to delete.
+	$path_blocked['ROOT']           = rtrim( ABSPATH, '/' );
+	$path_blocked['WP_CONTENT_DIR'] = WP_CONTENT_DIR;
+	$path_blocked['WP_ADMIN_DIR']   = ABSPATH . 'wp-admin';
+	$path_blocked['WP_INCLUDE_DIR'] = realpath( ABSPATH . WPINC );
+	$path_blocked                   = array_diff_key( $path_blocked, array_flip( $directories_unblock ) );
+	$file_folder_path               = realpath( ltrim( dirname( $file ), '/' ) );
+
+	if ( in_array( $file_folder_path, $path_blocked ) ) {
```

Comment:
Before doing this, you will need to normalize the slashes - forward `/` vs backward `\`, i.e. Unix vs Windows - will give problems in this comparison.

You also need to pass the third parameter, `$strict`, to `in_array()` as otherwise, this comparison is vulnerable to hacks.

Ref: http://php.net/manual/en/function.in-array.php

Reply:
Yeah, you right I didn't think about this. To resolved I implement realpath on all itens and I am use DIRECTORY_SEPARATOR, parameter of define slashes. done.